daemon/k8s/init.go

```diff
@@ -126,8 +126,8 @@ func WaitForNodeInformation(ctx context.Context, log logrus.FieldLogger, localNo
 			logfields.K8sNodeIP:        k8sNodeIP,
 		}).Info("Received own node information from API server")
 
-		// If the host does not have an IPv6 address, return an error
-		if option.Config.EnableIPv6 && nodeIP6 == nil {
+		// If the host does not have an IPv6 address and IPv4 is enabled, return an error
```

Comment:
nit: We can change the comment here to clarify the intention for throwing the error.
In case of a dual stack cluster, we throw an error if ipv6 is nil for a node. (To avoid a single stack cluster with dual stack config)

Reply:
I've tried to incorporate both the reasons for throwing an error in the dual-stack scenario as well as not throwing one in the single-stack scenario into the error message.